components/datetime/src/timezone.rs

```diff
@@ -0,0 +1,331 @@
+            _ => None,
+        })
+        .filter_map(|field| match field.symbol {
+            FieldSymbol::TimeZone(zone) => Some((u8::from(field.length), zone)),
+            _ => None,
+        })
+        .flat_map(|(length, zone_symbol)| match zone_symbol {
+            TimeZone::LowerZ => match length {
+                1..=3 => vec![Ok(provider::key::TIMEZONE_SPECIFIC_NAMES_SHORT_V1)],
+                4 => vec![Ok(provider::key::TIMEZONE_SPECIFIC_NAMES_LONG_V1)],
+                _ => vec![Err(DateTimeFormatError::Pattern(
+                    PatternError::FieldTooLong(FieldSymbol::TimeZone(zone_symbol)),
+                ))],
+            },
+            TimeZone::LowerV => match length {
+                1 => vec![
+                    Ok(provider::key::TIMEZONE_GENERIC_NAMES_SHORT_V1),
+                    Ok(provider::key::TIMEZONE_EXEMPLAR_CITIES_V1),
+                ],
+                4 => vec![
+                    Ok(provider::key::TIMEZONE_GENERIC_NAMES_LONG_V1),
+                    Ok(provider::key::TIMEZONE_EXEMPLAR_CITIES_V1),
+                ],
+                _ => vec![Err(DateTimeFormatError::Pattern(
+                    PatternError::FieldTooLong(FieldSymbol::TimeZone(zone_symbol)),
+                ))],
+            },
+            TimeZone::UpperV => match length {
+                1 => vec![], // BCP-47 identifier, no CLDR-data necessary.
+                2 => vec![], // IANA time-zone ID, no CLDR data necessary.
+                3 | 4 => vec![Ok(provider::key::TIMEZONE_EXEMPLAR_CITIES_V1)],
+                _ => vec![Err(DateTimeFormatError::Pattern(
+                    PatternError::FieldTooLong(FieldSymbol::TimeZone(zone_symbol)),
+                ))],
+            },
+            // ISO-8601 or localized GMT formats. CLDR data is either unneeded or required by default.
+            TimeZone::LowerX | TimeZone::UpperX | TimeZone::UpperZ | TimeZone::UpperO => vec![],
+        })
+        .collect()
+}
+
+/// Loads a resource into its destionation if the destionation has not already been filled.
+fn load_resource<'d, D, L, P>(
+    locale: &L,
+    resource_key: ResourceKey,
+    destionation: &mut Option<Cow<'d, D>>,
+    provider: &P,
+) -> Result<(), DateTimeFormatError>
+where
+    D: std::fmt::Debug + Clone + 'd,
+    L: Clone + Into<LanguageIdentifier>,
+    P: DataProvider<'d, D> + ?Sized,
+{
+    if destionation.is_none() {
+        *destionation = Some(
+            provider
+                .load_payload(&DataRequest {
+                    resource_path: ResourcePath {
+                        key: resource_key,
+                        options: ResourceOptions {
+                            variant: None,
+                            langid: Some(locale.clone().into()),
+                        },
+                    },
+                })?
+                .payload
+                .take()?,
+        );
+    }
+    Ok(())
+}
+
+pub struct TimeZoneFormat<'d> {
```

Comment:
Issue: Is this the main public TimeZoneFormat struct?  Please add documentation :)

Reply:
Certainly.

I was holding off on documenting in full, with examples, until I got a first pass from you and Zibi, in case you had major changes for me.

I've pushed up a commit responding to everything else, and I'm working on documentation right now.